### PR TITLE
disttask: add tests for dist role

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #48795 

Problem Summary:

### What changed and how does it work?
1. add tests for role.
2. refine stm to mgr in task_table.go
wait for #50108 
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by AilinKid — APPROVED** [review]

File: pkg/disttask/framework/storage/util.go [hunk 1]
```diff
@@ -104,11 +104,27 @@ func GetSubtaskEndTimeForTest(ctx context.Context, stm *TaskManager, subtaskID i
 	return time.Time{}, nil
 }
 
-// PrintSubtaskInfo log the subtask info by taskKey. Only used for UT.
-func (stm *TaskManager) PrintSubtaskInfo(ctx context.Context, taskID int64) {
-	rs, _ := stm.executeSQLWithNewSession(ctx,
+// GetSubtaskNodesForTest gets subtasks running nodes for one task for test.
+func GetSubtaskNodesForTest(ctx context.Context, mgr *TaskManager, taskID int64) ([]string, error) {
+	rs, err := mgr.executeSQLWithNewSession(ctx,
+		`select exec_id from mysql.tidb_background_subtask where task_key=%? group by exec_id`, taskID)
```
Comment (tangenta) [inline pkg/disttask/framework/storage/util.go:110]: It is not necessary to use `group by`, maybe `select distinct(exec_id)` is enough.

Comment (ywqzzy) [reply pkg/disttask/framework/storage/util.go]: https://github.com/pingcap/tidb/pull/50115/commits/1cbf435edd22be486cbeffda9492ff99efe03d76

**Review by tangenta — APPROVED** [review]

**Pull request merged**